Clarifying job merge feature (#7269)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/git/connect-gitlab.md

## For dbt Cloud Developer and Team tiers

:::note

> Only Team plan accounts will be able to configure [Git merges](/docs/deploy/merge-jobs) to trigger job runs. This feature is not available to Developer plans.

Reviewer: What do you think about clearly calling both plans here? I know this is in the Dev and Team tier plan subcategory but I totally misunderstood this until I saw the other section.
```suggestion
You can only configure Team and Enterprise plan accounts to enable [Git merges](/docs/deploy/merge-jobs) to trigger job runs. This feature is not available to Developer plans.
```

Author: Makes sense!
